### PR TITLE
TMEDIA-207 remove excess canary packages

### PR DESCRIPTION
**Pull request by JackHowa**
## Description
Programmatically Clear Unused Published Canary Releases In Blocks
Takes in the newest 100 versions of a package and matches to delete based on the canary text in the string

## Jira Ticket
- [TMEDIA-207](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-207)

## Acceptance Criteria

- You don't delete any stable nor beta releases 
- Every canary publish should clear out non-used ones 
- Should apply to every block

## Test Steps
- Look at previous run deleting one block's matching version
- Look at previous runs matching many blocks versions that should be deleted https://github.com/WPMedia/fusion-news-theme-blocks/runs/2569543175?check_suite_focus=true#step:2:53
- All blocks included in list. Note: there's a max of 20 per run. I haven't hit github rate-limiting yet. But the author of the package (which I forked) noted that the limit can be hit

### Before

- Growing amount of blocks with canary in their version. We should only need a few most recent ones. Hurdling toward the cliff of having too many packages for Github packages

### After
- See nonused block deleting "canary"-matching versions among newest 100 https://github.com/WPMedia/fusion-news-theme-blocks/runs/2570107225?check_suite_focus=true . five most recent matches not deleted for continuity of local development
- See the dry-run runs matching blocks (20 per step max for extension) but not deleting the five most recent https://github.com/WPMedia/fusion-news-theme-blocks/runs/2569543175?check_suite_focus=true#step:2:53 (note: this is a dry-run, no blocks actually deleted)

## Dependencies or Side Effects
- If you're using locally a canary block that gets deleted, I feel like that cached block may force you to re-run `npx fusion start -f`
- On push to canary, the github action is run pruning unused canary tags
- Had to fork a preexisting github action that was missing a feature we'd need (at least to start)https://github.com/SmartsquareGmbH/delete-old-packages 
- the official github action wouldn't work as we wanted https://github.com/actions/delete-package-versions

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block. -> ran in github actions 
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: .github/workflows/delete-unused-canary-packages.yml
Comment (RobinGiannattasio) [inline .github/workflows/delete-unused-canary-packages.yml:107]: So it looks like we have to add every block individually - can we add something into the documentation that upon creation of a new block, we should be adding it into this delete method?

File: .github/workflows/delete-unused-canary-packages.yml
Comment (RobinGiannattasio) [inline .github/workflows/delete-unused-canary-packages.yml]: Not that I don't want to keep you forever, but does it make sense to have this moved into WPMedia instead of hosting on your own github?

File: .github/workflows/delete-unused-canary-packages.yml
Comment (JackHowa) [reply .github/workflows/delete-unused-canary-packages.yml:107]: good call

File: .github/workflows/delete-unused-canary-packages.yml
Comment (JackHowa) [reply .github/workflows/delete-unused-canary-packages.yml]: hahah ya can do 😿 . I have a pr open with the maintainer so fingers-crossed they can merge my fork in. 

will update

Comment (RobinGiannattasio) [reply .github/workflows/delete-unused-canary-packages.yml]: never leave

Comment (JackHowa) [reply .github/workflows/delete-unused-canary-packages.yml]: hahahah. awaiting approval from it-help-desk for the github marketplace

File: .github/workflows/delete-unused-canary-packages.yml
Comment (JackHowa) [reply .github/workflows/delete-unused-canary-packages.yml:107]: updated https://github.com/WPMedia/fusion-news-theme-blocks/pull/864/files#diff-c330d9620da7e572515775e229f7b894bedd7c29b122a7295c6b6da4b0c21475R92 @RobinGiannattasio

**Review by matthewroach — DISMISSED** [review]
All looks good to me.. Shame we have to manually list all the blocks within the action workflow file

**Comment by JackHowa** [comment]
> Shame we have to manually list all the blocks within the action workflow file

Yeah I know. I looked into trying to apply this on a per org basis. But then we'd have to be able to ensure there wouldn't be side effects for non-blocks package naming schemes

File: .github/workflows/delete-unused-canary-packages.yml
Comment (JackHowa) [reply .github/workflows/delete-unused-canary-packages.yml]: updated 😿

**Review by RobinGiannattasio — APPROVED** [review]
LOVE the new block checklist. Thanks!

**Pull request merged**
